src/lib/transaction_snark/transaction_snark.ml

```diff
@@ -4187,25 +4191,42 @@ module For_tests = struct
       spec
     in
     let sender_pk = sender.public_key |> Public_key.compress in
-    let fee_payer =
-      { Party.Fee_payer.body =
-          { public_key = sender_pk
-          ; update = Party.Update.noop
-          ; token_id = ()
-          ; balance_change = fee
-          ; increment_nonce = ()
-          ; events = []
-          ; sequence_events = []
-          ; call_data = Field.zero
-          ; call_depth = 0
-          ; protocol_state_precondition =
-              Zkapp_precondition.Protocol_state.accept
-          ; account_precondition = sender_nonce
-          ; use_full_commitment = ()
+    let fee_payer : Party.Fee_payer.t =
+      match fee_payer_opt with
+      | None ->
```

Comment:
could you use the match case to get public key and nonce to avoid duplicating the other values?